src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    if (this.hidden) {
+      this.hidden = false;
+    } else {
+      this.hidden = true;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivoreAnimal) {
```

Comment:
```suggestion
  bite(animal) {
```
you don't know for sure it's an herbivore or not, that's why you check it below